src/hooks/privateChats.ts

```diff
@@ -215,3 +225,61 @@ export const useRecipientChat = (recipientId: string) => {
     recipient,
   };
 };
+
+export const usePrevious = <T>(value: T) => {
+  const ref = useRef<T>(value);
+  useEffect(() => {
+    ref.current = value;
+  });
+  return ref.current;
+};
+
+export const usePrivateChatNotification = (
+  privateChatMessages: WithId<PrivateChatMessage>[]
+) => {
+  const { user } = useUser();
+  const userId = user?.uid;
+
+  const { chatSettings } = useChatSidebarControls();
+
+  const [play] = useSound(newMessageSound, {
+    // `interrupt` ensures that if the sound starts again before it's
+    // ended, it will truncate it. Otherwise, the sound can overlap.
+    interrupt: true,
+  });
+
+  const privateChateMessagesSorted = useMemo(
+    () => Object.values(privateChatMessages).sort(chatSort),
+    [privateChatMessages]
+  );
+
+  const prevPrivateChatMessages = usePrevious(privateChateMessagesSorted[0]);
+
+  const currentChatUserId =
+    chatSettings.openedChatType === ChatTypes.PRIVATE_CHAT
+      ? chatSettings.recipientId
+      : undefined;
+
+  return useMemo(() => {
+    const shouldPlaySound = privateChateMessagesSorted
+      .filter((message) => {
+        return (
+          message?.ts_utc.seconds > prevPrivateChatMessages?.ts_utc.seconds
+        );
+      })
+      .some(
+        (message) =>
+          !message.isRead && ![userId, currentChatUserId].includes(message.from)
+      );
```

Comment:
I think we can probably combine the `.filter()` and `.some()` together here, since we're not actually storing the filtered results/using them elsewhere.

Something like:

```suggestion
      .some(
        (message) => {
          const isNewMessage = message?.ts_utc.seconds > previousNewestMessage?.ts_utc.seconds;
          const notRead = !message.isRead;
          const notFromMeOrCurrentChat = ![userId, currentChatUserId].includes(message.from);
          
          return isNewMessage && notRead && notFromMeOrCurrentChat;
      });
```